### ISSUE TITLE
Remove attachment from email

### ISSUE THREAD

**Issue by albogabriel** [pixelated/pixelated-user-agent#549]
## Definition
As a user who added an attachment to an email
I want to remove the attachment
So that I can opt not to send it.

## Constraints
This story depends on #548 being done.

## UI
When clicking on the X in the right corner of the attachment rectangle, the file must be removed from the email. No action confirmation will be required.

![image](https://cloud.githubusercontent.com/assets/350155/11749049/9cd81f0e-a012-11e5-922d-2535216e9a0b.png)

Note: Removing an attachment must make the body input occupy the space where the attachment was, as can be seen below:

![image](https://cloud.githubusercontent.com/assets/350155/11749269/df48d9f4-a013-11e5-8c1c-1b4cba723654.png)

**Comment by jeffhsta** [comment]
I implemented the remove attachment, it seems to be done, but I found out a bug that when I try forward an email with one or more attachment and if I remove an attachment when I am composing this forward email the original file attached to the received email will be gone.

That happens because the phash generated for the SoledadDocument is based on content file, so event if we duplicate the file, I think this bug will continue happing.

I think it's better create another issue to fix this bug on SoledadDocument before complete fully this issue

**Comment by felixhammerl** [comment]
sorry, but i'll revert this in most parts because of the aforementioned bug. here's the thing:
1. send yourself a message with an attachment
2. write a draft, upload the same attachment
3. close draft, navigate somewhere else.
4. go back to drafts, re-open draft, delete attachment
5. open the first message in sent, try downloading the attachment. **boom.**

**Comment by felixhammerl** [comment]
sorry :-/

**Comment by mnandri** [comment]
Given the current shortcoming of leap with regards to attachment deletion. 
The scope of this card is reduced to:
* User removing attachments as he composes a new email, reply, reply all, forward or edit drafts
* The attachment will be kept as a soledad doc and not deleted.

A new card will be created to work with leap on actual soledad deletion of an attachment.

For QA, please make sure that forwarding or editing drafts do not delete the original attachment that is already in the sent email... yet it will not be attached to the new forwarded/sent email.

@jeffhsta please add anything I forgot...

**Comment by pereiragislene** [comment]
Considering the scenario below:
* User removing attachments as he composes a new email, reply, reply all, forward or edit drafts
 - If you upload the same file more than once and delete one of them, the other files get deleted altogether. NOTE.: when uploading files with same name and extension (different files), deleting one of them doesn't affect the other.

**Comment by jeffhsta** [comment]
Hey @pereiragislene, the behavior should be, add or remove an attachment and that will change just on the JS part, but when saving the draft, it will change the back end part.

Also any changes on attachments should trigger the save drafts event.

Are you removing the drafts and clicking on send button faster than save drafts job?

**Comment by pereiragislene** [comment]
@jeffhsta: Not sure if we are talking about the same thing. The problem isn't the save draft job. Let me try to rephrase it.

These are the steps I took:

1. Compose a new mail
2. Choose a valid file and upload it 3 times 
3. The same file should appear 3 times in the UI
4. Delete one of the files on the UI
5. The other 2 files are gone
6. Send this email (apparently without any attachment) to yourself
7. Open the received mail. Note that it has 2 attachments.

I guess the problem is step 5 only.

**Comment by jeffhsta** [comment]
@pereiragislene: I think now I understood your point, I will try reproduce that.

**Comment by jeffhsta** [comment]
@pereiragislene I did a commit that should fix the bug you pointed out, could you check it?

**Comment by pereiragislene** [comment]
I tested it locally and it's working fine now :+1: 
Can be launched.
